examples/example_memfn.cpp

```diff
@@ -0,0 +1,35 @@
+#include <algorithm>
+#include <dice/template-library/memfn.hpp>
+
+#include <cassert>
+#include <ranges>
+#include <string>
+#include <vector>
+
+struct NumberProcessor {
+	std::vector<int> numbers_ = {1, 2, 3, 4, 5, 6, 7, 8};
+	std::string prefix_ = "item-";
+
+	static bool is_even(int n) { return n % 2 == 0; }
+
+	[[nodiscard]] std::string decorate(int n) const { return prefix_ + std::to_string(n); }
+
+	[[nodiscard]] std::string item_6() const { return "item-6"; }
+
+
+	void example() {
+		auto processed_view = numbers_ |
+							  std::views::filter(DICE_MEMFN(is_even)) |
```

Comment:
I think here `DICE_MEMFN` is not needed because `is_even` is a static member function